src/importexport/midi/internal/midiexport/exportmidi.cpp

```diff
@@ -402,6 +404,31 @@ bool ExportMidi::write(QIODevice* device, bool midiExpandRepeats, bool exportRPN
                     }
                 }
             }
+
+            // export RehearsalMarks only for first track
+            if (staffIdx == 0) {
+                for (Segment* seg = rs->firstMeasure()->first(); seg && seg->tick().ticks() < endTick; seg = seg->next1()) {
```

Comment:
Filtering for ChordRest and TimeTick segments should be enough, and would slightly increase performance:
```suggestion
                for (Segment* seg = rs->firstMeasure()->first(Segment::CHORD_REST_OR_TIME_TICK_TYPE); 
                     seg && seg->tick().ticks() < endTick; 
                     seg = seg->next1(Segment::CHORD_REST_OR_TIME_TICK_TYPE)) {
```